src/views/ReactionsView.tsx

```diff
@@ -0,0 +1,150 @@
+import { ReactElement, useState } from "react";
+import { Message, MessageReaction } from "../model/db";
+import { useReactions } from "../hooks/useReactions";
+import { Reaction, addReaction, removeReaction } from "../model/reactions";
+import { useClient } from "../hooks/useClient";
+import classNames from "classnames";
+
+const defaultReactions = [
+  ["👍", "thumbsup"],
+  ["👎", "thumbsdown"],
+  ["🎉", "tada"],
+];
+
+function nameToEmoji(name: string): string {
+  return {
+    thumbsup: "👍",
+    thumbsdown: "👎",
+    tada: "🎉",
+  }[name]!;
+}
+
+function ReactionView({
+  clientAddress,
+  name,
+  reactions,
+  onAdd,
+  onRemove,
+}: {
+  clientAddress: string;
+  name: string;
+  reactions: MessageReaction[];
+  onAdd: () => void;
+  onRemove: () => void;
+}): ReactElement {
+  const currentUserReacted = reactions
+    .map((reaction) => reaction.reactor)
+    .includes(clientAddress);
+
+  function onClick() {
+    if (currentUserReacted) {
+      onRemove();
+    } else {
+      onAdd();
+    }
+  }
+
+  return (
+    <button
+      onClick={onClick}
+      className={classNames(
+        "py-0.5 px-2 border rounded text-xs text-zinc-500",
+        {
+          "bg-yellow-200": currentUserReacted,
+        }
+      )}
+    >
+      {nameToEmoji(name)} {reactions.length}
+    </button>
+  );
+}
+
+export default function ReactionsView({
+  message,
+}: {
+  message: Message;
+}): ReactElement {
+  const client = useClient()!;
+  const reactions = useReactions(message) || [];
+  const [isReacting, setIsReacting] = useState(false);
+
+  const reactionsByName: { [key: string]: MessageReaction[] } =
+    reactions.reduce(
+      (acc: { [key: string]: MessageReaction[] }, curr: MessageReaction) => {
+        if (acc[curr.name]) {
+          acc[curr.name].push(curr);
+        } else {
+          acc[curr.name] = [curr];
+        }
+
+        return acc;
+      },
+      {}
+    );
+
+  return (
+    <div className="flex space-x-2">
+      {Object.keys(reactionsByName).map((name) => {
```

Comment:
might need some optional chaining for this one, in case the call in the hook fails and it stays as an empty obj.

Reply:
Would it just return an empty array and do nothing?

```ts
Object.keys({}) // => []
```

Reply:
yup! spoke too soon with that one, sorry -- opened this PR up in transit.